### PR TITLE
Include names in the shadowed subvolumes popup (bsc#928641)

### PR DESCRIPTION
**Pull request by ancorgs**

### CODE REVIEW

File: src/lib/storage/shadowed_vol_list.rb [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Yast.import "FileSystems"
+
+module Yast
+  class ShadowedVolList
+    include Enumerable
+    include Yast::Logger
+
+    attr_reader :partition, :mntpoint
+
+    def initialize(partition: nil, mntpoint: nil)
```
Comment (shundhammer) [inline src/lib/storage/shadowed_vol_list.rb]: NP: mntpoint is not really that intuitive for a named parameter. Any good reason against mount_point which would be better to memorize?

File: src/lib/storage/shadowed_vol_list.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      subvols = partition.fetch("subvol", [])
+      log.debug "Full list of subvolumes: #{subvols}"
+      subvols.select do |subvol|
+        name = full_name(subvol)
+        shadowed?(name)
+      end
+    end
+
+    def shadowed?(name)
+      (name == mntpoint) || (name.start_with?(mntpoint) && name[mntpoint.size] == "/")
```
Comment (shundhammer) [inline src/lib/storage/shadowed_vol_list.rb]: Not sure what exactly this expression does:

   name[mntpoint.size] == "/"

maybe it deserves a dedicated function with a speaking name?

File: src/lib/storage/shadowed_vol_list.rb [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    end
+
+    def full_name(subvol)
+      if FileSystems.default_subvol.empty?
+        "/" + subvol.fetch("name")
```
Comment (shundhammer) [inline src/lib/storage/shadowed_vol_list.rb]: subvol.fetch("name") is ycpkiller'ish for subvol["name"]?

File: src/include/partitioning/custom_part_check_generated.rb [hunk 4]
```diff
@@ -549,6 +549,12 @@ def check_created_partition_table(targetMap, installation)
       ok
     end
 
+    def format_subvols(subvols)
+      # Extra space needed for the regexp
+      all = subvols.join(', ') + " "
```
Comment (shundhammer) [inline src/include/partitioning/custom_part_check_generated.rb]: Not sure why this lone space is added at the end when later there is a strip() call

**Comment by shundhammer** [comment]
Aside from the small nitpicking issues LGTM.

File: src/lib/storage/shadowed_vol_list.rb (hunk 1 above)
Comment (ancorgs) [inline src/lib/storage/shadowed_vol_list.rb]: just typing laziness. I'll change it.

File: src/include/partitioning/custom_part_check_generated.rb (hunk 4 above)
Comment (ancorgs) [inline src/include/partitioning/custom_part_check_generated.rb]: Read the comment :-)

File: src/lib/storage/shadowed_vol_list.rb (hunk 3 above)
Comment (ancorgs) [inline src/lib/storage/shadowed_vol_list.rb]: Not really ycpkiller'ish. I use fetch often also in new code when I consider it looks better :-)

File: src/lib/storage/shadowed_vol_list.rb (hunk 2 above)
Comment (ancorgs) [inline src/lib/storage/shadowed_vol_list.rb]: Good point. I'll do it

**Pull request merged**